### PR TITLE
Proposal for release notes 4.5.0-rc

### PR DESCRIPTION
**Pull request by nagmat84**
IMHO they are still too long (TLTR). If anybody has suggestions to make it more succinct, feel welcome to push a commit.

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -32,6 +32,18 @@
 
 ## Version 4
 
+### v4.5.0-rc
+
+This is a _release candidate_ which only affects those users who follow the master branch immediately.
+Important things to know:
+
+- Despite the classification as a release candidate your data (photos and albums) should be as save as with any other release. Don't worry.
```
Comment (ildyria) [inline docs/releases.md:40]:
```suggestion
- Despite the classification as a release candidate your data (photos and albums) should be as safe as with any other release. Don't worry.
```

File: docs/releases.md [hunk 2]
```diff
@@ -30,6 +30,17 @@
 
 ## Master branch
 
+Lychee currently undergoes a major architectural refactoring of its internals.
```
Comment (kamil4) [inline docs/releases.md:33]: I'm wondering if _is currently undergoing_ wouldn't be better...

File: docs/releases.md [hunk 3]
```diff
@@ -30,6 +30,17 @@
...
+The master branch already includes the first part of that refactoring which mostly concentrates on the backend.
+
+These changes and information only affect those users who directly follow the master branch:
+
+- Despite the substantial changes your photos and albums should be as safe as with any other commit to the master branch. Don't worry.
```
Comment (kamil4) [inline docs/releases.md:38]: Honestly, I would advocate for more caution, such as recommending that users back up their DB before migrating. How about something like:

_While we have made every effort to ensure that your existing photos and albums are as safe as with any other commit to the master branch, due to the sheer size of current changes we recommend that you take additional precautions, such as creating a backup of your SQL database before migrating._

File: docs/releases.md [hunk 4]
```diff
@@ -30,6 +30,17 @@
...
+- As we did our best to ensure that Lychee behaves as before, you might potentially encounter more regression bugs than usually. These bugs are _not_ expected to severely affect the overall functioning of Lychee, but may be annoying.
```
Comment (kamil4) [inline docs/releases.md:39]: I'm not clear what message we are trying to convey with this sentence, so the below suggestion is just my guess:

_While we did our best to ensure that Lychee behaves as before, the large amount of recent changes means that you may encounter more regression bugs than usually. Please make sure to test the functionality you depend on and promptly report any regressions you observe; we will do our best to address any such bugs as quickly as possible._

File: docs/releases.md [hunk 5]
```diff
@@ -30,6 +30,17 @@
...
+- **New ID scheme:** Albums and photos are migrated to a new ID scheme. Previously, albums and photos used time-based integer IDs. The new IDs are truly 144bit randomness encoded in Base64 as a 24-character string, e.g., if your photo had the URL [https://lycheeorg.github.io/r/16102925744307/16102927818284](https://lycheeorg.github.io/demo/#16102925744307/16102927818284) before, it may have the ID [https://lycheeorg.github.io/r/GTqZfSso3nPeCnTNW4ovisgC/jM6KkmlK7X0LtVas5MjrHtTO](https://lycheeorg.github.io/demo/#16102925744307/16102927818284) after the migration. This means, **external links to your albums or photos will stop working.** However, we implemented a redirection service which _is disabled by default_. You can enable the re-direction service under `Settings` > `More` > `legacy_id_redirection`. This gives you time to migrate any external reference to the new IDs. We also generate a log entry every time the redirection service is used. This log contains the legacy and new ID as well as information where the request came from. This may help you with the migration of your external links.
```
Comment (kamil4) [inline docs/releases.md:40]: 144bit _of_ randomness (?)

File: docs/releases.md [hunk 6]
```diff
@@ -30,6 +30,17 @@
...
+- **Annoying quirks already known:** This is a list of already known regressions to be fixed by the second part of the refactoring which will concentrate on the JSON API and front-end:
+  - _Password-dialog is shown for non-existing albums:_ If a user requests a URL for a non-existing album, the password dialog will be presented to the user as if the album was password-protected. The password dialog will repeatedly re-presented to the user independent of the provided password as the authentication will never succeed.
```
Comment (kamil4) [inline docs/releases.md:42]: Unless you changed something in the meantime, it's not just non-existing albums that trigger it but also private ones.

will _be_ repeatedly re-presented

**Review by kamil4 — APPROVED** [review]
Overall I like it better than the first attempt but I would strike a different tone here and there...

File: docs/releases.md (hunk 3 above)
Comment (d7415) [reply docs/releases.md:38]: This seems sensible. It's also worth remembering we've assured users that master is stable in the past.

File: docs/releases.md (hunk 4 above)
Comment (d7415) [reply docs/releases.md:39]: I like @kamil4's version, but `s/usually/usual/`

File: docs/releases.md (hunk 5 above)
Comment (d7415) [reply docs/releases.md:40]: "The new IDs are 24-character base64-encoded strings." ?

Is it worth the URLs being tagged, especially if one of them is going to break immediately?

Also, I still lean towards enabling the redirect by default.

File: docs/releases.md (hunk 6 above)
Comment (d7415) [reply docs/releases.md:42]: "will be presented repeatedly"?

**Comment by nagmat84** [comment]
> Overall I like it better than the first attempt but I would strike a different tone here and there...

Any specific suggestions? I am not good at writing socially acceptable texts. I am usually straight forward and blunt.

So feel free to rewrite it. It is only a proposal.

**Comment by nagmat84** [comment]
Ah, sorry, I just saw now that you gave specific comments. The stupid Android full client does not show them. I only saw the overall comment.

The normal Github website in a mobile browser is better than the native full client. Why do the offer a full client if the only difference is that it hides things. 🤔

**Pull request merged**